src/components/storagePools/createStoragePoolDialog.jsx

```diff
@@ -211,6 +212,35 @@ const StoragePoolSourceRow = ({ onValueChanged, dialogValues }) => {
                 </FormGroup>
             </Grid>
         );
+    else if (dialogValues.type == 'fs')
+        return (
+            <Grid hasGutter>
+                <FormGroup fieldId='storage-pool-dialog-source' label={_("Source path")}
+                           className="pf-m-8-col"
+                           id="storage-pool-dialog-source-group">
+                    <FileAutoComplete id='storage-pool-dialog-source'
+                                      superuser='try'
+                                      placeholder={placeholder}
+                                      onChange={value => onValueChanged('source', { device: value })} />
+                    <FormHelper fieldId="storage-pool-dialog-source" helperTextInvalid={validationState == "error" && _("Source path should not be empty")} />
+                </FormGroup>
+                <FormGroup fieldId='storage-pool-dialog-source-format' label={_("Format")}
+                           className="pf-m-4-col">
+                    <FormSelect id='storage-pool-dialog-source-format'
+                                value={dialogValues.source.format}
+                                onChange={(_event, value) => onValueChanged('source', { format: value })}>
+                        { diskPoolSourceFsTypes
```

Comment:
This whole new block almost perfectly duplicates the existing `disk` case, except for this list. Please merge them check for "disk" || "fs", and do this here:
```
dialogValues.type === 'fs' ? diskPoolSourceFsTypes : diskPoolSourceFormatTypes)
```

Reply:
Thank you for your advice!
Merged two blocks.
After the change, it seems to work fine.